actioncable/lib/action_cable/channel/base.rb

```diff
@@ -144,13 +144,14 @@ def initialize(connection, identifier, params = {})
 
         # When a channel is streaming via pubsub, we want to delay the confirmation
         # transmission until pubsub subscription is confirmed.
-        @defer_subscription_confirmation = false
+        #
+        # We use atomic fixnum to track the number of waiting tasks to avoid race conditions
```

Comment:
Like the one below, this comment isn't pulling its weight: here it just seems to be describing the line of code. Let's change it to instead focus on the thing the assignment doesn't tell us:

```ruby
#
# The counter starts at 1 because it's awaiting a call to #subscribe_to_channel.
```